### PR TITLE
feat(ctb): use SCREAMING_SNAKE_CASE for immutables

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Makes sure we use SCREAMING_SNAKE_CASE for immutables everywhere. Adds legacy getters where necessary.

**Tests**
Existing tests should cover.

### CODE REVIEW

File: packages/contracts-bedrock/contracts/L1/L1CrossDomainMessenger.sol [hunk 1]
```diff
@@ -17,7 +17,8 @@ contract L1CrossDomainMessenger is CrossDomainMessenger, Semver {
     /**
      * @notice Address of the OptimismPortal.
      */
-    OptimismPortal public immutable portal;
+    // solhint-disable-next-line var-name-mixedcase
+    OptimismPortal public immutable PORTAL;
```
Comment (smartcontracts) [inline packages/contracts-bedrock/contracts/L1/L1CrossDomainMessenger.sol:21]: Only some places need legacy getters. This was a new variable as of Bedrock, so no legacy getter required.

**Review by maurelian — APPROVED** [review]
I think you could also make these immutable in this PR, otherwise LGTM

https://github.com/ethereum-optimism/optimism/blob/f80f01a3ea23ff5ab2c2e3f55314dfbec9eb7afd/packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol#L19-L24

**Review by tynes — APPROVED** [review]
Meh at legacy getters, but understandable

**Comment by tynes** [comment]
Re: the `op-bindings` issue, we did fix the issue at one point by having only one compiler version used for compiling erc20

**Comment by smartcontracts** [comment]
@mergifyio requeue

**Comment by smartcontracts** [comment]
@mergifyio requeue

**Pull request merged**